### DISCUSSION TITLE
fire request in a conditional statement

### DISCUSSION THREAD

**Discussion by Xi-gofore** [TanStack/query#1899] [General] [answered]
I am not sure is it my misunderstanding or not, but in general if I want to fire a request when certain condition is met, how can I do it? My impression of react-query seems like only support fire request when component is loaded.

In my case I only want to fire request to backend if condition is met:

```
const fireReqOnlyIfConditionMet = async ({myData}) => {
     if(myData.length>10) {
       // how can I fire request here ? Say it is a POST request, I need to use useMutation()
     }
}
```

**Comment by TkDodo** [comment, marked as answer]
you are talking about `useMutation`, right? This returns a `mutate` function, so the idea is to unconditionally call `useMutation`, this doesn't trigger a request. it gives you a `mutate` function, and once you invoke this function (can be in a condition), the mutation is started.

**Reply by Xi-gofore** [reply]
Thanks for clarifying. I misunderstood it.

**Reply by Xi-gofore** [reply]
Could you please take a look at a related question here https://github.com/tannerlinsley/react-query/discussions/1904 ?
